src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php

```diff
@@ -486,4 +486,34 @@ public function getLocalKeyName()
     {
         return $this->localKey;
     }
+
+    /**
+     * Get the foreign attributes.
+     *
+     * @return array
+     */
+    public function getForeignAttributes()
+    {
+        return [$this->getForeignKeyName() => $this->getParentKey()];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function insert(array $values)
+    {
+        if (empty($values)) {
+            return true;
```

Comment:
Curious: why return true here? In my own code, I would throw an Exception here, but I imagine for adding to the framework, it should return false?

Reply:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Query/Builder.php#L3261-L3272

Reply:
Interesting. Never noticed that before. Thanks!